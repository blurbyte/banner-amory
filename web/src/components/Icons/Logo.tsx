import * as React from 'react';

class Logo extends React.Component {
  render() {
    return (
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 66 90" width="66" height="90" {...this.props}>
        <path d="M50 90L25 67.5 0 90V0h50v90z" fill="#d92f29" />
        <path
          d="M58.8 37.4l-1.5-19.7c0-.3 0-.5.1-.8.4-.9 1.3-1 1.6-1l1.3-.1-.4-5.7H47.4c-.7-.7-1.5-1.3-2.3-1.9L42 6l-1 3.6c0 .1-.1.3-.1.4H15.7l1.9 4.2v.1c.2.4.4.9.6 1.5.6 1.5 1 3 1.1 4.3.2 1.6.1 2.9-.3 4-.4.9-1.1 1.6-2.1 2.1-.6.2-1.3.4-1.8.5-.5.1-.9.1-1.4.1h-.5c-.6-.1-.7-.2-.7-.2v-.1c1.2 0 2.3-.6 2.9-1.6.9-1.5.5-3.6-1-4.6-.5-.4-1.2-.6-1.8-.6-1.2 0-2.3.6-2.9 1.7-.6 1-1.4 2.7-1.5 4.6-.1 2 .8 3.8 2.4 4.7 1 .6 2.2.8 3.7.8.4 0 .8 0 1.3-.1 1.8-.2 3.2-.5 4.5-1.2 1.5-.8 2.7-1.9 3.5-3.4 1.5-2.8 1.7-6.3.7-10.8h4.4c-.1 3-.2 5.4-.5 7.7-.4 3.2-.9 7-2 10.5-1 3.4-2.4 6.3-4.4 8.9-.9 1.3-2 2.5-3.2 3.9-.5.6-1.1.9-1.9.9-.7 0-1.2-.3-1.2-.3l-.8-.4-2.7 2.5 8.4 8.5.9-1c3.2-3.5 5.6-6.6 7.5-9.6 1.7-2.7 3-5.7 3.9-9h11.5l.5 6.7c.2 3.2-.6 5.2-2.7 6-1.1.4-1.6.5-3.3.5h-3.9v5.8h16.7c.5 2.6 1.7 3.5 2 3.7l.8.5.7-.6c.1-.1 1.3-1.2 1.5-3.6h9.2l.1-1.2c.6-7.4-1.8-13.8-7-19zm-24.5-7.8c.2-1.6.3-3.4.3-5.4.1-3 .2-5.5.2-8.2H39.9c-.3 5 .8 9.6 3.3 13.6h-8.9zm25.4 19.7l-.1-1.6c.4 1.1.8 2.3 1 3.5-.6-.5-.8-1.1-.9-1.9z"
          fill="#ffdbac"
        />
        <path
          fill="#222222"
          d="M64.6 56.4H56l-1.3-16.6c1.2 1.1 2.3 2.3 3.2 3.5l.5 6c.2 2.6 1.8 3.3 3.6 3.6-.4-5.5-2.8-10.2-7.4-14.1l-.3-3.5c7.5 5.5 10.9 12.5 10.3 21.1zm-8.5-39.8l-.2.2c-1.8 1.9-2.6 5.4-2.2 10.1l.6 7.7c1.1.8 2.1 1.6 3.1 2.5L56 17.8c0-.4 0-.8.1-1.2zm-.6-.3c1.3-1.4 2.8-1.6 3.4-1.7l-.2-3.2H47.8c2.2 2.3 4.3 5.7 5.3 11 .3-2.7 1.1-4.7 2.4-6.1zm-5.6 11.1c-.6-7.7-2.9-11.4-5.1-13.5-1 6.7.7 12.3 5.4 17.1l-.3-3.6zm1.6 21.8c-.3.9-.7 1.7-1.2 2.3-1.4 1.5-3.4 1.7-5 1.7h-9.1v3.2h15.9l-.6-7.2zm-1.7-4.6c-.8 0-1.5.7-1.5 1.5s.7 1.5 1.5 1.5 1.4-.6 1.5-1.4V46c-.1-.8-.7-1.4-1.5-1.4zm0 3.6c-1.2 0-2.1-.9-2.1-2.1 0-1.2.9-2.1 2.1-2.1.5 0 1 .2 1.4.5l-.6-7.9c-2.2-1.8-4.1-3.7-5.6-5.7H32.9c.3-2.3.4-4.5.5-6.8.1-3.4.2-6.1.2-9.5h5.2c.7 0 1.3 0 1.9.1.1-1.1.3-2.2.5-3.3H17.7l1.1 2.3c.2.5.5 1 .7 1.6.6 1.5 1 3 1.2 4.6.2 1.5.2 3.2-.3 4.6-.5 1.3-1.5 2.3-2.8 2.8-.7.3-1.4.4-2.1.5-.8.2-1.6.2-2.3.2-.7-.1-1.4-.3-1.7-.9-.3-.7-.2-1.6 0-2.3 1 .7 2.3.3 2.9-.7.6-1 .3-2.2-.6-2.8-1-.7-2.3-.3-3 .7-.7 1.2-1.3 2.6-1.3 4-.1 1.4.5 2.8 1.8 3.5 1.3.7 2.9.7 4.3.6 1.4-.1 2.7-.4 4-1 1.3-.6 2.3-1.6 3-2.9 1.6-3 1.3-6.7.6-9.9-.1-.5-.3-1.1-.4-1.6h7.4c0 2.9-.1 6-.5 9.2-.5 3.7-1 7.3-2 10.7-1 3.4-2.5 6.5-4.5 9.3-1 1.4-2.1 2.7-3.3 4-2.1 2.2-4.6.9-4.6.9l-1.1 1.1 6.5 6.7c2.7-2.9 5.2-5.9 7.3-9.4 1.8-3 3.2-6.3 4-9.6h13.7l.6 7.9c.3 4.5-1.4 6.4-3.5 7.3h2.7c1.5 0 3.3-.2 4.6-1.5.7-.8 1.2-1.9 1.3-3.4-.6.1-1.1.3-1.6.3zm5.5 8.2l-2.2-29.2c-.9-10.8-5-15.1-8.8-17.9l-1.5-1.1-.5 1.8c-2.7 10.4 0 19 8.2 25.7l-.3-3.7c-5.5-5.3-7.4-11.7-5.9-19.4 3 2.6 5.5 6.5 6.2 14.8l2.2 29c.3 3.1 1.6 3.9 1.6 3.9s1.2-1 1-3.9z"
        />
      </svg>
    );
  }
}

export default Logo;
