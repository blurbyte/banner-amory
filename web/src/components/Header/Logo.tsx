import styled from 'styled-components';

import { Logo as BaseLogo } from '../Icons';

const Logo = styled(BaseLogo)`
  position: absolute;
  z-index: 9999;
  top: 0;
  left: 2rem;
`;

export default Logo;
