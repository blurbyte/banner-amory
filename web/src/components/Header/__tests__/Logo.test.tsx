import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Logo from '../Logo';

jest.mock('../../Icons', () => ({
  Logo: 'BaseLogo'
}))

test('renders correctly', () => {
  const component = renderer.create(<Logo />);
  expect(component.toJSON()).toMatchSnapshot();
});
