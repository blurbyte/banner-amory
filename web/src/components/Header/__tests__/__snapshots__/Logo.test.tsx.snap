// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: absolute;
  z-index: 9999;
  top: 0;
  left: 2rem;
}

<BaseLogo
  className="c0"
/>
`;
